x/evm/keeper/fees.go

```diff
@@ -55,8 +57,26 @@ func (k *Keeper) DeductTxCostsFromUserBalance(
 		return errorsmod.Wrapf(err, "account not found for sender %s", from)
 	}
 
+	// Convert the balance to 18 decimals
+	signerBalance := k.bankKeeper.GetBalance(ctx, signerAcc.GetAddress(), utils.BaseDenom)
```

Comment:
we should not use `utils.BaseDenom` here but rather get EVM denom from the EVM params